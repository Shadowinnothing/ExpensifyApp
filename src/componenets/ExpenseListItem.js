import React from 'react';
import {connect} from 'react-redux';

import {removeExpense} from '../actions/expenses';

const ExpenseListItem = ({dispatch, description, amount, createdAt, id}) => (
  <div>
    <h3>Description: {description}</h3>
    <p>${amount} - {createdAt}</p>
    <br />
    <button
      onClick={() => {
        dispatch(removeExpense({id}))
      }}
    >Remove</button>
  </div>
);

export default connect()(ExpenseListItem);
