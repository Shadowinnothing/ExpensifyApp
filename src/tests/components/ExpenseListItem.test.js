import React from 'react';
import {shallow} from 'enzyme';

import ExpenseListItem from '../../components/ExpenseListItem';
import expenses from '../fixtures/expenses';

describe('components/ExpenseListItem', () => {

  test('should render single ExpenseListItem', () => {
    const wrapper = shallow(<ExpenseListItem {...expenses[0]} />);
    expect(wrapper).toMatchSnapshot();
  });

});
