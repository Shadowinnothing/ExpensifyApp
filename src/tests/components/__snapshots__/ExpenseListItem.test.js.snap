// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ExpenseListItem should render single ExpenseListItem 1`] = `
<Link
  className="list-item"
  replace={false}
  to="/edit/1"
>
  <div>
    <h3
      className="list-item__title"
    >
      Gum
    </h3>
    <span
      className="list-item__sub-title"
    >
      December 31st, 1969
    </span>
  </div>
  <div>
    <h3
      className="list-item__data"
    >
      $3.50
    </h3>
  </div>
</Link>
`;
